### PR TITLE
DEPR: Enforce DTI.to_series(keep_tz) removal

### PR DESCRIPTION
**Pull request by mroeschke**
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Introduced in https://github.com/pandas-dev/pandas/pull/29731

### CODE REVIEW

File: pandas/core/indexes/datetimes.py [hunk 1]
```diff
@@ -565,29 +544,7 @@ def to_series(self, keep_tz=lib.no_default, index=None, name=None):
...
-                    FutureWarning,
-                    stacklevel=find_stack_level(inspect.currentframe()),
-                )
-        else:
-            keep_tz = True
-
-        if keep_tz and self.tz is not None:
+        if self.tz is not None:
             # preserve the tz & copy
             values = self.copy(deep=True)
```
Comment (jbrockmendel) [inline pandas/core/indexes/datetimes.py:549]: Can we do self.copy in both cases?

Comment (mroeschke) [reply pandas/core/indexes/datetimes.py:549]: Looks like there's a copy in both if/else branches. Did you mean use `deep=True` in both cases?

Comment (jbrockmendel) [reply pandas/core/indexes/datetimes.py:549]: right.  so L592-L596 could just be `values = self._values.copy()` i think

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @mroeschke (please update the deprecation issue)

**Pull request merged**